include/dlaf/auxiliary/norm/mc.h

```diff
@@ -78,21 +80,25 @@ dlaf::BaseType<T> Norm<Backend::MC, Device::CPU, T>::max_L(comm::CommunicatorGri
       else
         return lange(lapack::Norm::Max, tile);
     });
-    auto current_tile_max = pika::dataflow(norm_max_f, matrix.read(tile_wrt_local));
+    auto current_tile_max =
+        matrix.read_sender(tile_wrt_local) |
+        dlaf::internal::transform(dlaf::internal::Policy<Backend::MC>(), std::move(norm_max_f)) |
+        ex::make_future();
 
     tiles_max.emplace_back(std::move(current_tile_max));
   }
 
-  // than it is necessary to reduce max values from all ranks into a single max value for the matrix
-
-  // TODO unwrapping can be skipped for optimization reasons
-  NormT local_max_value = pika::dataflow(unwrapping([](const auto&& values) {
-                                           if (values.size() == 0)
-                                             return std::numeric_limits<NormT>::min();
-                                           return *std::max_element(values.begin(), values.end());
-                                         }),
-                                         tiles_max)
-                              .get();
+  // then it is necessary to reduce max values from all ranks into a single max value for the matrix
+
+  auto max_element = [](std::vector<NormT>&& values) {
+    DLAF_ASSERT(!values.empty(), "");
+    return *std::max_element(values.begin(), values.end());
+  };
+  NormT local_max_value =
+      tiles_max.empty() ? std::numeric_limits<NormT>::min()
+                        : sync_wait(ex::when_all_vector(std::move(tiles_max)) |
+                                    dlaf::internal::transform(dlaf::internal::Policy<Backend::MC>(),
+                                                              std::move(max_element)));
```

Comment:
I don't know if it was deliberate or not, but `sync_wait` is not using the "pipeline style".

Reply:
It is actually deliberate! I haven't checked if it's "non-pipeline" everywhere now, but changed the ones that were touched by this PR otherwise.

`pika::this_thread::sync_wait` (as you've noticed) can be used at the end of a pipeline. However, P2300 actually doesn't allow that, so I thought I'd start changing them to "non-pipeline" already now.